app/views/devise/sessions/_mfa_enrolment.erb

```diff
@@ -1,29 +1,32 @@
-<h1 class="govuk-heading-xl"><%= t('mfa_enrolment.heading') %></h1>
-
 <div class="govuk-grid-row">
-  <div class="govuk-grid-column-one-half">
-
-      <%= f.label :totp_code, class: "govuk-label" %>
-      <%= f.text_field :totp_code, autofocus: true, autocomplete: 'off', class: "govuk-input govuk-input--width-10" %>
-      <%= f.hidden_field :email, value: { email: false } %>
-
-  </div>
-
-  <div class="govuk-grid-column-one-half">
-    <div class="mfa-qr-code">
-      <%= raw(@secret_code_svg) %>
-    </div>
+  <div class="govuk-grid-column-full">
+    <h1 class="govuk-heading-xl"><%= t('mfa_enrolment.heading') %></h1>
+      <p class="govuk-body-m"><%= t('mfa_enrolment.description') %></p>
+        <div style="text-align:left;">
+          <%= raw(@secret_code_svg) %>
+        </div>
+      <p class="govuk-body-m"> 
+        <details class="govuk-details" data-module="govuk-details">
+          <summary class="govuk-details__summary">
+            <span class="govuk-details__summary-text">
+              <%= t('mfa_enrolment.no_qr') %>
+            </span>
+          </summary>
+          <div class="govuk-details__text">
+            <%= t('mfa_enrolment.no_qr_instructions') %>
+            <pre><%= @secret_code %></pre>
+          </div>
+        </details>
+      </p>
+      <div>
+        <div class="govuk-!-margin-bottom-7">
+          <h1 class="govuk-heading-s"><%= t('mfa_enrolment.heading_code_entry') %></h1>
```

Comment:
```suggestion
          <h4 class="govuk-heading-s"><%= t('mfa_enrolment.heading_code_entry') %></h4>
```
there's already ona h1 and this heading is not on the same level, but a sub-heading.

https://design-system.service.gov.uk/styles/typography/#headings